### ISSUE TITLE
[REQ] add preformatted text dataset support

### ISSUE THREAD

**Issue by sozercan** [kaito-project/aikit#821]
Parent: #214

Depends on #820 and #598.

## Summary

Add a `text` dataset type for preformatted, full-sequence causal-language-model training.

```yaml
datasets:
  - source: organization/domain-corpus
    type: text
```

Expected record:

```json
{"text": "A complete preformatted training sequence."}
```

AIKit must own the sequence-boundary contract so every record has one effective leading BOS where applicable and one terminal EOS.

## Motivation

Preformatted text is common for domain corpora, source code, documents, already-rendered instruction examples, and custom language-modeling sequences.

The frozen Unsloth text path tokenizes `text`, but does not provide the complete public contract AIKit needs:

- EOS is not appended automatically.
- Unsloth 2026.8.1 can duplicate an existing BOS token.
- Tokenizers differ in whether BOS is represented in text or added during tokenization.

This feature is full-sequence text SFT. It must not be described as complete continued pretraining, which needs different LoRA targets and optimizer configuration.

## Proposed changes

- Add a `text` dataset type constant and runtime dispatch.
- Require a non-empty dataset and non-empty string `text` values.
- Add tokenizer-aware normalization:
  - Require a usable EOS token.
  - Append EOS when absent.
  - Do not duplicate terminal EOS.
  - Avoid duplicate effective BOS after tokenization.
  - Do not prepend BOS when the tokenizer adds it automatically.
- Preserve source content other than special-token boundary normalization.
- Continue using `SFTTrainer` with full-sequence loss.
- Ensure packing operates on explicitly EOS-terminated records.
- Add source/row context to validation errors where available.
- Add fixtures, documentation, and a one-step GPU smoke configuration.

## Acceptance criteria

- `type: text` is accepted by Go validation and Python dispatch.
- Text records never use the Alpaca formatter.
- Every normalized record has exactly one terminal EOS.
- Existing EOS is not duplicated.
- Existing or tokenizer-added BOS is not duplicated.
- Tokenizers without BOS remain supported; a tokenizer without EOS fails clearly.
- Null, empty, or non-string values fail with useful errors.
- Full-sequence labels supervise the normalized record.
- Packing retains explicit record boundaries.
- Existing Alpaca behavior remains unchanged.
- A one-step GPU run exports a non-empty GGUF.

## Tests

- Go validation tests for `type: text`.
- Python normalization tests covering missing/existing EOS, explicit/automatic/no BOS, missing EOS token, and invalid values.
- Exact-lock regression proving no duplicate effective BOS and one terminal EOS.
- Add:
  - `test/fixtures/text-smoke.jsonl`
  - `test/aikitfile-unsloth-text-smoke.yaml`
- Run `make test`, `make test-python`, `make lint`, and a GPU fine-tune/export smoke test.

## Out of scope

- Proper continued pretraining.
- Training `embed_tokens` or `lm_head`.
- An embedding-specific learning rate.
- Pretokenized datasets.
- Messages, ShareGPT, or response-only loss.
- Multiple datasets or streaming.
- DPO, GRPO, vision, or audio.
